artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -315,6 +315,37 @@ public void onNotification(final Notification notification) {
 
                queueInfos.put(clusterName, info);
 
+               if (distance < 1) {
+                  //Binding added locally. If a matching remote binding with consumers exist, add a redistributor
+                  Binding binding = getBinding(routingName);
+
+                  if (binding != null) {
+                     try {
+                        Bindings bindings = getBindingsForAddress(address);
+
+                        for (Binding bind : bindings.getBindings()) {
+                           if (bind.isConnected() && bind instanceof RemoteQueueBinding) {
+
+                              RemoteQueueBinding remoteBinding = (RemoteQueueBinding) bind;
+
+                              if (remoteBinding.consumerCount() > 0) {
+
+                                 Queue queue = (Queue) binding.getBindable();
+                                 AddressSettings addressSettings = addressSettingsRepository.getMatch(binding.getAddress().toString());
+                                 long redistributionDelay = addressSettings.getRedistributionDelay();
+
+                                 if (redistributionDelay != -1) {
```

Comment:
I wonder if we want to break earlier if this condition is not met rather than checking each remoteBinding consumer count etc. maybe check the addressSettings at line:319

Reply:
Yes, that makes a lot of sense, I will look into that.